cql3/expr/expression.cc

```diff
@@ -1304,6 +1304,10 @@ constant evaluate(term* term_ptr, const query_options& options) {
         return constant::make_null();
     }
 
+    if (dynamic_cast<abstract_marker*>(term_ptr) != nullptr) {
+        return evaluate(term_ptr->to_expression(), options);
+    }
```

Comment:
I don't understand this. Why do we even have evaluate(term*)? Shouldn't we convert to expressions during prepare, then call evaluate(expression)?

This is going to be very slow, term->expression conversion is not supposed to be in a fast path.

Reply:
Sure

Reply:
Not so sure. We may branch 4.6 in between, leaving 4.6 with the regression.